docs/Makefile

```diff
@@ -2,7 +2,7 @@
 #
 
 # You can set these variables from the command line.
-SPHINXOPTS    = -W --keep-going -n -jauto
```

Comment:
I'm pretty sure we want to keep this in the future, so it fails loudly when we have legitimate errors. As you pointed out, we have #11972 to track this, and there's an open PR to fix it in #12044. That being said, I need to review that PR soon before I can review yours, so we can fix both problems. Will do when I find the time... 🙂